### PR TITLE
Hide irreleveant command line options of lldb-moduleimport-test

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Comment by adrian-prantl** [comment]
@swift-ci smoke test

File: tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp [hunk 1]
```diff
@@ -181,30 +181,38 @@ int main(int argc, char **argv) {
   INITIALIZE_LLVM();
 
   // Command line handling.
-  llvm::cl::list<std::string> InputNames(
-    llvm::cl::Positional, llvm::cl::desc("compiled_swift_file1.o ..."),
-    llvm::cl::OneOrMore);
+  using namespace llvm::cl;
+  static OptionCategory Visible("Specific Options");
+  HideUnrelatedOptions({&Visible});
```
Comment (jrose-apple) [inline tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: It looks like `HideUnrelatedOptions(None)` will hide everything, and then the following options don't need to get a category.

Comment (adrian-prantl) [reply tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: Good to know! But this way it also prints nicely in the --help output as

General options:
  --version ...
Specific options:
  --dump-module ...

Comment (adrian-prantl) [reply tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: By the way, do you happen to know what's up with the failure in https://ci.swift.org/job/swift-PR-osx/8961/ ?

Comment (jrose-apple) [reply tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: @graydon's disabling the test in #20581 while he investigates.

**Comment by adrian-prantl** [comment]
Oh. I didn't realize that that test failure was connected to this change! From the looks of it, `module-cache-diagnostics.swift` doesn't even run `lldb-moduleimport-test` — do you understand how they are connected?

**Comment by adrian-prantl** [comment]
Oh I think I misread your comment: They aren't!

**Pull request merged**